internal/apis/configuration/v1alpha1/types.go

```diff
@@ -0,0 +1,680 @@
+	// http://nginx.org/en/docs/http/ngx_http_access_module.html
+	WhitelistSourceRange []net.IPNet `json:"whitelist-source-range"`
+
+	// Limits the rate of response transmission to a client.
+	// The rate is specified in bytes per second. The zero value disables rate limiting.
+	// The limit is set per a request, and so if a client simultaneously opens two connections,
+	// the overall rate will be twice as much as the specified limit.
+	// http://nginx.org/en/docs/http/ngx_http_core_module.html#limit_rate
+	LimitRate int `json:"limit-rate"`
+
+	// Sets the initial amount after which the further transmission of a response to a client will be rate limited.
+	// http://nginx.org/en/docs/http/ngx_http_core_module.html#limit_rate_after
+	LimitRateAfter int `json:"limit-rate-after"`
+}
+
+// Log ...
+type Log struct {
+	// EnableAccessLog enables the Access Log globally from NGINX ingress controller
+	//http://nginx.org/en/docs/http/ngx_http_log_module.html
+	EnableAccessLog bool `json:"enable-access-log"`
+
+	// http://nginx.org/en/docs/ngx_core_module.html#error_log
+	// Configures logging level [debug | info | notice | warn | error | crit | alert | emerg]
+	// Log levels above are listed in the order of increasing severity
+	ErrorLogLevel string `json:"error-log-level"`
+
+	// FormatEscapeJSON enables json escaping
+	// http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format
+	FormatEscapeJSON bool `json:"format-escape-json"`
+
+	// FormatUpstream customize upstream log_format
+	// http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format
+	FormatUpstream string `json:"format-upstream"`
+
+	// FormatStream customize stream log_format
+	// http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format
+	FormatStream string `json:"format-stream"`
+
+	// SkipAccessLogURLs sets a list of URLs that should not appear in the NGINX access log
+	// This is useful with urls like `/health` or `health-check` that make "complex" reading the logs
+	// By default this list is empty
+	SkipAccessLogURLs []string `json:"skip-access-log-urls"`
+
+	File   LogFileConfiguration `json:"file"`
+	Syslog SyslogConfiguration  `json:"syslog"`
+}
+
+// LogFileConfiguration ...
+type LogFileConfiguration struct {
+	// AccessLogPath sets the path of the access logs if enabled
+	// http://nginx.org/en/docs/http/ngx_http_log_module.html#access_log
+	// By default access logs go to /var/log/nginx/access.log
+	AccessLogPath string `json:"access-log-path"`
+
+	// ErrorLogPath sets the path of the error logs
+	// http://nginx.org/en/docs/ngx_core_module.html#error_log
+	// By default error logs go to /var/log/nginx/error.log
+	ErrorLogPath string `json:"error-log-path"`
+}
+
+/*
+Latency = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10}
+ResponseLength = []float64{100, 1000, 10000, 100000, 1000000}
+RequestLength = []float64{1000, 10000, 100000, 1000000, 10000000}
+*/
+
+type Metrics struct {
```

Comment:
Do the variables here require a json definition?

Reply:
Yes, to allow `kubectl get nginxconfiguration XXXXX -o yaml`

Reply:
👍